src/librustc_parse/parser/diagnostics.rs

```diff
@@ -225,8 +225,21 @@ impl<'a> Parser<'a> {
             self.token.span,
             &format!("expected identifier, found {}", self.this_token_descr()),
         );
+        let valid_follow = &[
+            TokenKind::Eq,
+            TokenKind::Colon,
+            TokenKind::Comma,
+            TokenKind::Semi,
+            TokenKind::ModSep,
+            TokenKind::OpenDelim(token::DelimToken::Brace),
```

Comment:
I think this should also include `Bracket`-- e.g.:

```rust
macro_rules! async { () => {} }
```

Reply:
(but it is a heuristic, and as such it may be better not to suggest changing `async { ...  }`)